app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -28,7 +28,8 @@ def context
     end
 
     def user
-      @user ||= User.find_by(omniauth_params.slice(:provider, :uid).to_h)
+      @user ||= User.find_by(omniauth_params.slice(:provider, :uid).to_h) ||
+                User.find_by(email: omniauth_params['info']['email'])
```

Comment:
We should set `uid` and `provider` on the user that was found for future lookups :)